proposals/NNNN-Add-screen-stack-management-module.md

```diff
@@ -0,0 +1,96 @@
+- Add `button Name` for subscribe.
+- Add screen change notification process `notifyScreenChanged` to SDL app library.
+- Add the screen management module to [the iOS Guides](https://smartdevicelink.com/en/guides/iOS/displaying-a-user-interface/template-images/), [the Android Guides](https://smartdevicelink.com/en/guides/android/displaying-a-user-interface/main-screen-templates/) and [the JavaScript Guides](https://smartdevicelink.com/en/guides/javascript/displaying-a-user-interface/main-screen-templates/).
+
+Below is the `Button Name` for subscribe.
+Add element to enum value ButtonName. (HMI_API.xml, MOBILE_API.xml)
+
+```xml
+ <enum name="ButtonName" since="1.0">
+     <description>Defines the hard (physical) and soft (touchscreen) buttons available from the module</description>
++    <element name="MENU" />
++    <element name="BACK" />
+ </enum>
+
+```
+
+The sequence is as follows.
+
+1. Subscribe to `BACK` on the Back button.
+2. Subscribe to `MENU` on the menu button.
+3. The app displays MEDIA screen with `setDisplayLayout`. ScreenTaskManager detects `newScreen` and stacks MEDIA screen.
+4. The app displays the GRAPHIC_WITH_TILES screen with `setDisplayLayout`. ScreenTaskManager detects `keepLevel`, deletes the top layer screen of the screen stack, and stacks GRAPHIC_WITH_TILES screen.
+5. The user pushes the menu button. Then, head unit (HU) sends `OnButtonPress (MENU)` to the app.
+6. The app displays the Menu screen with `showAppMenu`. ScreenTaskManager detects `newScreen` and stacks the menu screen.
+7. The user pushes the submenu button. Then, the HU sends the `OnCommand` to the app.
+8. The app displays submenu screen with `showAppMenu`. ScreenTaskManager detects `newScreen` and stacks the submenu screen.
+9. The user pushes the back button. Then, the HU sends `OnButtonPress (BACK_KEY)` to the app.
+10. The app returns to menu screen with `showAppMenu`.
+11. The app displays PerformInteraction screen with `setPerformInteraction`. ScreenTaskManager  stacks PerformInteraction screen.
+12. The app hides PerformInteraction screen with `setPerformInteraction`, and displays menu svreen. ScreenTaskManager deletes PerformInteraction screen of the screen stack.
+
+![new_sequence.png](../assets/proposals/NNNN-Add-screen-stack-management-module/new_sequence.png)
+
+## Potential downsides
+None.
+
+## Impact on existing code
+
+- A new module is added to the app library, therefore, it is affected as well.
+- Since there are new enum elements added, it affects SDL Core.
+- It is necessary to modify the following documentation.
+1. [the iOS Guides](https://smartdevicelink.com/en/guides/iOS/displaying-a-user-interface/template-images/)
+2. [the Android Guides](https://smartdevicelink.com/en/guides/android/displaying-a-user-interface/main-screen-templates/)
+3. [the JavaScript Guides](https://smartdevicelink.com/en/guides/javascript/displaying-a-user-interface/main-screen-templates/)
+
```

Comment:
Please also include JavaSE and JavaEE Guide links here if needed.

Reply:
Hi, @theresalech -san.
Thank you in advance for your prompt reply.
I think those are the target.
Therefore, I will modify this sentence to "SDL app library's documentation" and make a commit it.
Could you please condirm it.
By the way, I do not know what devices are JavaSE and JavaEE implemented.
So, could you please teach me those target devices.
Thank you.

Reply:
I've provided a suggested change to this line.  Please see my other comment for links to the JavaSE and JavaEE guides.  Thank you!